### ISSUE TITLE
Issue 007: fine infer が physics_payload なしでも center NPZ から出力 artifact を作れるようにする

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#340]
# Issue 007: fine infer が physics_payload なしでも center NPZ から出力 artifact を作れるようにする

## 背景

Issue 005-006 により、fine inference の local window center は `center_npz_files` と `window_center.npz_key` で generic に読めるようになる。しかし `run_fine_infer()` の後段はまだ physics artifact を必須としている。

現状:

```python
if physics_payload is None:
    physics_payload = load_physics_npz(typed.paths.physics_npz_files[0])
...
fine_infer_payload = build_fbpick_fine_infer_payload(
    coarse_payload=coarse_payload,
    physics_payload=physics_payload,
    fine_payload=fine_payload,
    ...
)
```

`coarse GPT => fine` を試すには、`center_npz_files = *.coarse.npz` の場合でも fine result を保存できる必要がある。

physics の存廃は未決定なので、既存 physics 経由の artifact schema は壊さない。

## 目的

fine infer が以下の2モードで動くようにする。

### Mode A: physics 経由、既存 default

```yaml
paths:
  center_npz_files:
    - line001.physics.npz
  physics_npz_files:
    - line001.physics.npz  # legacy / optional alias
  coarse_npz_files:
    - line001.coarse.npz
window_center:
  npz_key: physics_window_center_i
  valid_mask_npz_key: physics_window_valid_mask
```

### Mode B: coarse center direct

```yaml
paths:
  center_npz_files:
    - line001.coarse.npz
  coarse_npz_files:
    - line001.coarse.npz
window_center:
  npz_key: coarse_pick_i
  valid_mask_npz_key: coarse_center_valid_mask
```

Mode B では `physics_npz_files` がなくても `run_fine_infer()` が完走する。

## 非目的

- `FINE_INFER_REQUIRED_KEYS` の削除 / schema major version bump はしない。
- physics algorithm の削除はしない。
- final workflow を coarse=>fine に固定する判断はしない。

## 変更対象

```text
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/fine/infer.py
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/common/io.py
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/common/artifacts.py  # optional
packages/seisai-engine/tests/test_fbpick_fine_infer.py  # なければ既存 fine infer tests に追加
```

## 実装方針

既存 `build_fbpick_fine_infer_payload()` は physics_payload 必須なので、その直前で **synthetic physics-like payload** を作る helper を追加するのが最小変更。

```python
def build_synthetic_physics_payload_from_center_npz(
    *,
    center_payload: dict[str, np.ndarray],
    coarse_payload: dict[str, np.ndarray],
    center_i: np.ndarray,
    center_valid_mask: np.ndarray,
    center_prob: np.ndarray | None,
    lineage,
) -> dict[str, np.ndarray]:
    ...
```

または `common/io.py` に置く。

## synthetic physics payload 仕様

`build_fbpick_fine_infer_payload()` が要求する fields を満たす。

```text
dt_sec
n_samples_orig
n_traces
ffid_values
chno_values
offsets_m
trace_indices
physical_center_i
physical_center_t_sec
physical_model_status
physical_model_failure_reason
physical_offset_source
physical_runtime_fit_source
physics_conf
physics_source
physics_window_center_i
physics_window_center_t_sec
physics_window_valid_mask
lineage
```

値の決め方:

```text
metadata fields:
  coarse_payload からコピーする

physical_center_i:
  center_i を int32 で入れる

physical_center_t_sec:
  center_i * dt_sec。invalid は nan。

physics_window_center_i:
  center_i を int32 で入れる

physics_window_center_t_sec:
  center_i * dt_sec。invalid は nan。

physics_window_valid_mask:
  center_valid_mask

physics_conf:
  center_prob があれば clip(center_prob, 0, 1)
  なければ center_valid_mask True の場所は 1.0、False は 0.0

physics_source:
  PHYSICS_SOURCE_COARSE_OBSERVED

physical_model_status:
  互換用 sentinel。既存 validation が許す uint8 にする。
  推奨は zeros uint8。

physical_model_failure_reason:
  zeros uint8

physical_offset_source:
  zeros uint8

physical_runtime_fit_source:
  zeros uint8
```

注意: `load_physics_npz()` の strict validation を通す必要はない。`build_fbpick_fine_infer_payload()` の `_validate_physics_payload_for_fine_infer()` を通ればよい。

## center_prob の選び方

`window_center` に probability key を足す案もあるが、この issue では config 拡張を最小にする。

推奨実装:

1. `window_center.prob_npz_key` を optional 追加する。
2. default は `null`。
3. coarse direct example では `coarse_pmax` を指定できる。

```yaml
window_center:
  npz_key: coarse_pick_i
  valid_mask_npz_key: coarse_center_valid_mask
  prob_npz_key: coarse_pmax
```

`FineWindowCenterCfg` に追加:

```python
prob_npz_key: str | None = None
```

`_load_window_centers_for_info()` は centers / valid_mask だけ返しているため、prob は `run_fine_infer()` で center payload から読むか、dataset loader に返り値を増やす必要がある。実装を小さくするなら `prob_npz_key` はこの issue では optional metadata として `run_fine_infer()` 側で再読込してよい。

ただし acceptance は `prob_npz_key=None` でもよい。prob 対応はできれば入れる。

### 1. center payload を読み込む

`typed.paths.center_npz_files[0]` を generic loader で読む。

```python
center_payload = load_npz_payload(typed.paths.center_npz_files[0], payload_name='fine center npz')
```

### 2. physics payload があるか判定

```python
has_physics_input = typed.paths.physics_npz_files is not None
```

ただし `center_npz_files` が physics artifact の場合、`physics_npz_files` が省略されている可能性がある。安全側には以下の helper を使う。

```python
def _center_payload_looks_like_physics(payload) -> bool:
    return all(key in payload for key in ('physical_center_i', 'physics_conf', 'physics_source', 'physics_window_center_i'))
```

分岐:

```text
if physics_payload is provided:
    use it
elif paths.physics_npz_files is present:
    load_physics_npz(paths.physics_npz_files[0])
elif center_payload looks like physics:
    use center_payload as physics_payload, but validate via _validate_physics_payload_for_fine_infer or build_fbpick_fine_infer_payload
else:
    build synthetic physics-like payload from center payload + fine_payload.center_raw_i + fine_window_valid_mask
```

`synthesized` の場合、`physics_window_center_i` は `fine_payload['center_raw_i']` と一致させる。

### 3. coarse payload resolution

coarse payload は引き続き必要。

- `paths.coarse_npz_files` がある場合はそれを読む。
- ない場合、center payload が coarse-like なら center payload を coarse_payload として使える。
- ない場合で physics path から infer する legacy path logic は維持。

Mode B では `center_npz_files == coarse_npz_files` が多いので、同じ path を二重に読まないようにしてもよいが必須ではない。

## fine_pick_valid_mask の扱い

既存 `build_fbpick_fine_infer_payload()` は以下を使う。

```python
fine_pick_valid_mask = fine_window_valid_mask & physics_window_valid_mask & offset_valid_mask & ...
```

synthetic payload では `physics_window_valid_mask = center_valid_mask` なので、direct coarse center でも valid mask が正しく効く。

## viewer payload

`save_gather_png` では既存 overlay key を使っている。

```python
for key in ('physical_center_i', 'physics_window_center_i'):
    if key in physics_payload:
        viewer_payload[key] = np.asarray(physics_payload[key])
```

synthetic payload にもこの key があるため、既存 viewer は動く。

追加でわかりやすくするなら、fine output に以下を optional 保存してもよい。

```text
window_center_source_npz_path
window_center_npz_key
window_center_valid_mask_npz_key
```

ただし schema validation に入れる必要はない。

## テスト要件

候補テスト:

```python
def test_run_fine_infer_accepts_center_npz_without_physics_npz_files() -> None: ...

def test_build_synthetic_physics_payload_from_center_npz_matches_center_raw_i() -> None: ...

def test_fine_infer_direct_coarse_center_sets_fine_pick_valid_mask_from_center_valid_mask() -> None: ...

def test_fine_infer_still_uses_real_physics_payload_when_available() -> None: ...
```

model は既存 fine infer tests の dummy model pattern を使う。なければ、入力 shape と同じ time logits を返す最小 `torch.nn.Module` を用意する。

coarse-like center NPZ は最低限以下を含める。

```text
dt_sec
n_samples_orig
n_traces
ffid_values
chno_values
offsets_m
trace_indices
coarse_pick_i
coarse_pmax
coarse_center_valid_mask
lineage
coarse_pick_t_sec
coarse_prob_summary
```

`load_coarse_npz()` を使う場合は `COARSE_REQUIRED_KEYS` を満たすこと。

## 受け入れ条件

- `paths.center_npz_files` が `.coarse.npz` で、`paths.physics_npz_files` がない fine infer が完走する。
- 既存 physics 経由 fine infer は出力 schema / key を維持する。
- `fine_pick_valid_mask` は center valid mask を反映する。
- viewer gather PNG path で synthetic payload が原因の KeyError が出ない。

## 推奨確認コマンド

```bash
pytest packages/seisai-engine/tests -q -k 'fine and infer'
pytest packages/seisai-engine/tests/test_fbpick_physics_window_constraint.py -q
```

## 注意

この issue は physics を削除しない。direct coarse center を **評価可能にするための互換層** を入れるだけ。
